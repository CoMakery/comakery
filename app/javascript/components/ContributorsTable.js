import React from 'react'
import PropTypes from 'prop-types'

class ContributorsTable extends React.Component {
  render() {
    return (
      <React.Fragment>
        <div className='table-scroll table-box contributors'>
          <table className='table-scroll' style={{width: '100%'}}>
            <tbody>
              <tr className='header-row'>
                <th>
                  Contributors
                </th>
                <th>
                  Total Tokens Awarded
                </th>
              </tr>

              {this.props.tableData.map((t) =>
                <tr className='award-row' key={t.name}>
                  <td className='contributor'>
                    <img src={t.imageUrl} className='icon avatar-img' />
                    <div className='margin-small margin-collapse inline-block'>
                      {t.name}
                      <table className='table-scroll table-box overlay' style={{display: 'none'}}>
                        <tbody>
                          <tr>
                            <th style={{paddingBottom: '20px'}}>
                              Contribution Summary
                            </th>
                          </tr>

                          {t.awards.map((a) =>
                            <tr key={a.name}>
                              <td>
                                {a.name}
                              </td>
                              <td>
                                {a.total}
                              </td>
                            </tr>
                          )}
                        </tbody>
                      </table>
                    </div>
                  </td>

                  <td className='awards-earned financial'>
                    <span className='margin-small'>{t.total}</span>
                  </td>
                </tr>
              )}
            </tbody>
          </table>
        </div>

      </React.Fragment>
    )
  }
}

ContributorsTable.propTypes = {
  tableData: PropTypes.array.isRequired
}
ContributorsTable.defaultProps = {
  tableData: []
}
export default ContributorsTable
