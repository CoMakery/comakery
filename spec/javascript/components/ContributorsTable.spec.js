import React from 'react'
import { shallow } from 'enzyme'
import ContributorsTable from 'components/ContributorsTable'

describe('Contributors Table', () => {
  it('renders component correctly without props', () => {
    const wrapper = shallow(<ContributorsTable />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders component correctly without revenue share', () => {
    const tableData = [
      {
        'imageUrl': '/test.png',
        'name'    : 'Joe',
        'awards'  : [
          {
            'name' : 'Idea leading to patent',
            'total': '7,000'
          }
        ],
        'total': '0'
      },
      {
        'imageUrl': '/test2.png',
        'name'    : 'Betsy',
        'awards'  : [
          {
            'name' : 'Bugfix',
            'total': '20,000'
          },
          {
            'name' : 'Contribution',
            'total': '5,333'
          }
        ],
        'total': '0'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders component correctly with revenue share', () => {
    const tableData = [
      {
        'imageUrl': '/test2.png',
        'name'    : 'Betsy',
        'awards'  : [
          {
            'name' : 'Bugfix',
            'total': '20,000'
          },
          {
            'name' : 'Contribution',
            'total': '5,333'
          }
        ],
        'total'    : '49,877',
        'remaining': '49,377',
        'unpaid'   : '$1,348.68',
        'paid'     : '$15.08'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} revenueShare />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows table with USD', () => {
    const tableData = [
      {
        'imageUrl' : '/test2.png',
        'name'     : 'Betsy',
        'awards'   : [],
        'total'    : '49,877',
        'remaining': '49,377',
        'unpaid'   : '$1,348.68',
        'paid'     : '$15.08'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} revenueShare />)
    expect(wrapper.text()).toContain('Contributors')
    expect(wrapper.exists('td.contributor')).toBe(true)
    expect(wrapper.find('td.contributor').first().text()).toContain('Betsy')
    expect(wrapper.find('td.awards-earned').first().text()).toContain('49,877')
    expect(wrapper.find('td.paid').first().text()).toContain('$15.08')
    expect(wrapper.find('td.award-holdings').first().text()).toContain('49,377')
    expect(wrapper.find('td.holdings-value').first().text()).toContain('$1,348.68')
  })

  it('shows table with Bittoken', () => {
    const tableData = [
      {
        'imageUrl' : '/test2.png',
        'name'     : 'Betsy',
        'awards'   : [],
        'total'    : '0',
        'remaining': '0',
        'unpaid'   : '฿0',
        'paid'     : '฿0'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} revenueShare />)
    expect(wrapper.find('td.paid').first().text()).toContain('฿0')
    expect(wrapper.find('td.holdings-value').first().text()).toContain('฿0')
  })

  it('shows table with Eth', () => {
    const tableData = [
      {
        'imageUrl' : '/test2.png',
        'name'     : 'Betsy',
        'awards'   : [],
        'total'    : '0',
        'remaining': '0',
        'unpaid'   : 'Ξ0',
        'paid'     : 'Ξ0'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} revenueShare />)
    expect(wrapper.find('td.paid').first().text()).toContain('Ξ0')
    expect(wrapper.find('td.holdings-value').first().text()).toContain('Ξ0')
  })

  it('shows table without revenue share', () => {
    const tableData = [
      {
        'imageUrl' : '/test2.png',
        'name'     : 'Betsy',
        'awards'   : [],
        'total'    : '49,877',
        'remaining': '49,377',
        'unpaid'   : '$1,348.68',
        'paid'     : '$15.08'
      }
    ]
    const wrapper = shallow(<ContributorsTable tableData={tableData} />)
    expect(wrapper.text()).toContain('Contributors')
    expect(wrapper.exists('td.contributor')).toBe(true)
    expect(wrapper.find('td.contributor').first().text()).toContain('Betsy')
    expect(wrapper.find('td.awards-earned').first().text()).toContain('49,877')
    expect(wrapper.exists('td.paid')).toBe(false)
    expect(wrapper.exists('td.award-holdings')).toBe(false)
    expect(wrapper.exists('td.holdings-value')).toBe(false)
    expect(wrapper.text()).not.toContain('$')
    expect(wrapper.text()).not.toContain('Paid')
  })

  it('hides table without contributors', () => {
    const tableData = []
    const wrapper = shallow(<ContributorsTable tableData={tableData} />)
    expect(wrapper.exists('td.contributor')).toBe(false)
    expect(wrapper.exists('td.awards-earned')).toBe(false)
    expect(wrapper.exists('td.paid')).toBe(false)
    expect(wrapper.exists('td.award-holdings')).toBe(false)
    expect(wrapper.exists('td.holdings-value')).toBe(false)
  })
})