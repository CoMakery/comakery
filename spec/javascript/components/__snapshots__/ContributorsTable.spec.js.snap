// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributors Table renders component correctly with revenue share 1`] = `
<Fragment>
  <div
    className="table-scroll table-box contributors"
  >
    <table
      className="table-scroll"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <tbody>
        <tr
          className="header-row"
        >
          <th>
            Contributors
          </th>
          <th>
            Total Tokens Awarded
          </th>
          <th>
            Unpaid Revenue Shares
          </th>
          <th>
            Unpaid Revenue Share Balance
          </th>
          <th>
            Lifetime Paid
          </th>
        </tr>
        <tr
          className="award-row"
          key="Betsy"
        >
          <td
            className="contributor"
          >
            <img
              className="icon avatar-img"
              src="/test2.png"
            />
            <div
              className="margin-small margin-collapse inline-block"
            >
              Betsy
              <table
                className="table-scroll table-box overlay"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <tbody>
                  <tr>
                    <th
                      style={
                        Object {
                          "paddingBottom": "20px",
                        }
                      }
                    >
                      Contribution Summary
                    </th>
                  </tr>
                  <tr
                    key="Bugfix"
                  >
                    <td>
                      Bugfix
                    </td>
                    <td>
                      20,000
                    </td>
                  </tr>
                  <tr
                    key="Contribution"
                  >
                    <td>
                      Contribution
                    </td>
                    <td>
                      5,333
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </td>
          <td
            className="awards-earned financial"
          >
            <span
              className="margin-small"
            >
              49,877
            </span>
          </td>
          <td
            className="award-holdings financial"
          >
            <span
              className="margin-small"
            >
              49,377
            </span>
          </td>
          <td
            className="holdings-value financial"
          >
            <span
              className="margin-small"
            >
              $1,348.68
            </span>
          </td>
          <td
            className="paid hidden financial"
          >
            <span
              className="margin-small"
            >
              $15.08
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</Fragment>
`;

exports[`Contributors Table renders component correctly without props 1`] = `
<Fragment>
  <div
    className="table-scroll table-box contributors"
  >
    <table
      className="table-scroll"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <tbody>
        <tr
          className="header-row"
        >
          <th>
            Contributors
          </th>
          <th>
            Total Tokens Awarded
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</Fragment>
`;

exports[`Contributors Table renders component correctly without revenue share 1`] = `
<Fragment>
  <div
    className="table-scroll table-box contributors"
  >
    <table
      className="table-scroll"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <tbody>
        <tr
          className="header-row"
        >
          <th>
            Contributors
          </th>
          <th>
            Total Tokens Awarded
          </th>
        </tr>
        <tr
          className="award-row"
          key="Joe"
        >
          <td
            className="contributor"
          >
            <img
              className="icon avatar-img"
              src="/test.png"
            />
            <div
              className="margin-small margin-collapse inline-block"
            >
              Joe
              <table
                className="table-scroll table-box overlay"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <tbody>
                  <tr>
                    <th
                      style={
                        Object {
                          "paddingBottom": "20px",
                        }
                      }
                    >
                      Contribution Summary
                    </th>
                  </tr>
                  <tr
                    key="Idea leading to patent"
                  >
                    <td>
                      Idea leading to patent
                    </td>
                    <td>
                      7,000
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </td>
          <td
            className="awards-earned financial"
          >
            <span
              className="margin-small"
            >
              0
            </span>
          </td>
        </tr>
        <tr
          className="award-row"
          key="Betsy"
        >
          <td
            className="contributor"
          >
            <img
              className="icon avatar-img"
              src="/test2.png"
            />
            <div
              className="margin-small margin-collapse inline-block"
            >
              Betsy
              <table
                className="table-scroll table-box overlay"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <tbody>
                  <tr>
                    <th
                      style={
                        Object {
                          "paddingBottom": "20px",
                        }
                      }
                    >
                      Contribution Summary
                    </th>
                  </tr>
                  <tr
                    key="Bugfix"
                  >
                    <td>
                      Bugfix
                    </td>
                    <td>
                      20,000
                    </td>
                  </tr>
                  <tr
                    key="Contribution"
                  >
                    <td>
                      Contribution
                    </td>
                    <td>
                      5,333
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </td>
          <td
            className="awards-earned financial"
          >
            <span
              className="margin-small"
            >
              0
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</Fragment>
`;
